### PR TITLE
`IconTile` documentation update

### PR DESCRIPTION
**Pull request by andgen404**
An accessibility audit of the HVS application revealed confusion between interactive elements (like buttons) and non-interactive Icon Tiles. To address this, it is recommended that the Icon Tile documentation be updated with guidance on ensuring clear visual distinction and avoiding placement of Icon Tiles near buttons or other interactive elements.

### :pushpin: Summary

If merged, this PR will update the `IconTile` documentation with guidance on avoiding placement near interactive elements to improve distinguishability and accessibility. It will also update the illustrations with the visually updated `IconTile`. 

Link to Documentation ---------> https://hds-website-git-icontile-documentation-update-hashicorp.vercel.app/components/icon-tile#near-interactive-elements

### :link: External links

Jira ticket: [HDS-3586](https://hashicorp.atlassian.net/browse/HDS-3586)

***
[HDS-3586]: https://hashicorp.atlassian.net/browse/HDS-3586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 1]
```diff
@@ -58,4 +58,20 @@ A secondary icon can be added to provide additional context. For example, for an
 
 ## Using with content
 
-Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
\ No newline at end of file
+Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
+
+## Near interactive elements
+
+Avoid placing Icon Tiles next to buttons or other clickable elements. This can confuse users and make it unclear what is interactive. In these cases, use a simple icon for better usability.
```
Comment (majedelass) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:65]: I tried rewording this so that it can be a little more descriptive (plus providing a source) as to why we want folks to be mindful of this.

```suggestion
Avoid placing Icon Tiles next to visually similar interactive elements. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable) is an accessibility standard that deters interfaces from using elements that look the same but behave differently in close visual proximity. This can cause user confusion through the established visual language. 
```

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 2]
```diff
@@ -58,4 +58,20 @@ A secondary icon can be added to provide additional context. For example, for an
 
 ## Using with content
 
-Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
\ No newline at end of file
+Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
+
+## Near interactive elements
```
Comment (majedelass) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:63]: To follow the standard from the previous heading, maybe rename this to:

```suggestion
## Using with interactive elements
```

**Review by majedelass — COMMENTED** [review]
This is great, just had two notes. Feel free to word smith the first suggestion if you feel it's too wordy.

I think after we can open this up to the team.

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 3]
```diff
@@ -60,9 +60,9 @@ A secondary icon can be added to provide additional context. For example, for an
 
 Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
 
-## Near interactive elements
+## Using with interactive elements
 
-Avoid placing Icon Tiles next to buttons or other clickable elements. This can confuse users and make it unclear what is interactive. In these cases, use a simple icon for better usability.
+Avoid placing Icon Tiles next to visually similar interactive elements. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable) is an accessibility standard that deters interfaces from using elements that look the same but behave differently in close visual proximity. This can cause user confusion through the established visual language.
```
Comment (jorytindall) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:65]: I might suggest merging the last sentence into one of the previous points, otherwise it seems a bit like a fragment. I'm also not sure about "established visual language", that seems like an ambiguous statement. What about this:

```suggestion
Avoid placing Icon Tiles adjacent to visually similar interactive elements, as this can lead to user confusion due to lack of visual differentiation. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity. 
```

However, it bears mentioning that if we're recommending consumers not place Icon Tiles next to interactive elements, is the visual language of the Icon Tile to blame? does it appear _too_ interactive?

**Review by jorytindall — COMMENTED** [review]
One small note, probably brings up some larger questions though. I'll also through out that the example used is also referenced in the [Table content](https://helios.hashicorp.design/components/table?tab=content#product-branding) documentation. Not necessarily a problem, just noting the overlap, or maybe there's another example that could be used?

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md (hunk 3 above)
Comment (KristinLBradley) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md:65]: I think it would be worthwhile to explore a design update to the IconTile to make it less "interactive" looking. (Perhaps trying a solid color background without a border to differentiate more from Buttons?)

This is something that could be considered as a follow on though vs as part of this current PR/ticket.

Comment (andgen404) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md:65]: Yes, I agree. IconTile's visual design is worth discussing, especially considering Kristin's feedback. We should explore potential visual changes to make it appear less interactive.

Comment (andgen404) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md:65]: > Not necessarily a problem, just noting the overlap, or maybe there's another example that could be used?
I'll spin up another example, but I feel the one Melanie supplied in the Jira ticket makes the most sense as an example since I haven't seen many others where an IconTile is in that close proximity to our second button.

Comment (majedelass) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md]: FYI, played around with approach: https://www.figma.com/design/3VZiWIbcxIahVfVohQzhPk/Majed's-Playground?node-id=478-1032&t=4ZgbQi4ElepOVdVk-1

Ultimately the problem is that we use the same visual language as our buttons (border and shadow) which makes it similar. If we strip them, it'll look less interactive.

Thoughts @andgen404 ?

Comment (KristinLBradley) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md]: Just a mention from the meeting, I'd also try using a solid color vs. a gradient to see how that works.

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 4]
```diff
@@ -58,4 +58,20 @@ A secondary icon can be added to provide additional context. For example, for an
...
+
+!!! Dont
+
+![Dont example of Icon Tile placed in the same proximity of other interactive elements](/assets/components/icon-tile/icon-tile-example.png)
+!!!
+
+!!! Do
+
+![Do example of icon placed in the same proximity of other interactive elements, instead of an Icon Tile](/assets/components/icon-tile/icon-tile-example-do.png)
+!!!
```
Comment (andgen404) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:75]:
```suggestion

```

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 5]
```diff
@@ -58,4 +59,14 @@ A secondary icon can be added to provide additional context. For example, for an
 
 ## Using with content
 
-Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
\ No newline at end of file
+Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
+
+## Using with interactive elements
+
+Avoid placing Icon Tiles adjacent to visually similar interactive elements, as this can lead to user confusion due to lack of visual differentiation. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity.
```
Comment (majedelass) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:66]: We should reword this a bit to say something along the lines of:

```suggestion
Avoid placing Icon Tiles immediately next to secondary icon only button variants. Even though they visually look different, their proximity can confuse users because they share similar characteristics. [Distinguishability](https://www.w3.org/WAI/WCAG21/Understanding/distinguishable), an accessibility standard, discourages interfaces from using elements that look alike but behave differently when in close visual proximity. 
```

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 6]
```diff
@@ -6,56 +6,47 @@ To represent an object or as part of a page title.
 
 ### When not to use
 
-Within a paragraph where an object is referred to.
+- Within a paragraph where an object is referred to.
```
Comment (jorytindall) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:9]: I'm not really sure what you mean by this, can you clarify a bit?

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 7]
```diff
@@ -6,56 +6,47 @@ To represent an object or as part of a page title.
...
-  <Hds::IconTile @color="neutral" @icon="user" @size="medium" @iconSecondary="plus"/>
-  <Hds::IconTile @color="neutral" @icon="user" @size="small" @iconSecondary="plus"/>
-</Doc::Layout>
+![Icon Tiles with secondary icon](/assets/components/icon-tile/icon-tiles-scondary-icon.png)
 
 ## Using with content
 
-Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
\ No newline at end of file
+Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
```
Comment (jorytindall) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:42]: [suggestion] not opinionated on this, but this could be a way to combine the final sentence with the rest of the content, it seems like it's a bit of a fragment currently.

```suggestion
Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent as decorative element or accessory.
```

**Review by jorytindall — COMMENTED** [review]
I know this is still draft status, but took another quick pass with the changes you made!

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md (hunk 6 above)
Comment (andgen404) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md:9]: I can remove this snippet of text. I think what I was trying to say is, "Don't use outside of headings and standalone visuals." in a vague way 😅

File: website/docs/components/icon-tile/partials/guidelines/guidelines.md [hunk 8]
```diff
@@ -6,56 +6,46 @@ To represent an object or as part of a page title.
...
-  <Hds::IconTile @color="neutral" @icon="user" @size="medium" @iconSecondary="plus"/>
-  <Hds::IconTile @color="neutral" @icon="user" @size="small" @iconSecondary="plus"/>
-</Doc::Layout>
+![Icon Tiles with secondary icon](/assets/components/icon-tile/icon-tiles-scondary-icon.png)
 
 ## Using with content
 
-Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent. We recommend thinking of them as an accessory.
\ No newline at end of file
+Icon Tiles should not appear without accompanying content. Instead, they should be presented alongside a text label for the object or page they are being used to represent as decorative element or accessory.
```
Comment (KristinLBradley) [inline website/docs/components/icon-tile/partials/guidelines/guidelines.md:41]:
```suggestion
Use Icon Tiles as a decorative element or accessory next to a text label for the object or page they are being used to represent. Icon Tiles should not be used without accompanying content.
```

Comment (KristinLBradley) [reply website/docs/components/icon-tile/partials/guidelines/guidelines.md:41]: The current sentence is difficult to parse. I attempted to simplify and clarify in the above suggestion but feel free to tweak a bit more...
